Source/SwiftLintFramework/Extensions/SyntaxMap+SwiftLint.swift

```diff
@@ -12,18 +12,11 @@ extension SyntaxMap {
                 .intersects(byteRange)
         }
 
-        func notIntersect(_ token: SyntaxToken) -> Bool {
-            return !intersect(token)
-        }
-
         guard let startIndex = tokens.index(where: intersect) else {
             return []
         }
         let tokensBeginningIntersect = tokens.lazy.suffix(from: startIndex)
-        if let endIndex = tokensBeginningIntersect.index(where: notIntersect) {
-            return Array(tokensBeginningIntersect.prefix(upTo: endIndex))
-        }
-        return Array(tokensBeginningIntersect)
+        return Array(tokensBeginningIntersect.filter(intersect))
```

Comment:
I had to change this because it's not guaranteed that the first token that doesn't intersect the range is the last one that does. For example:

```json
[
  {
    "length" : 10,
    "offset" : 50,
    "type" : "source.lang.swift.syntaxtype.string"
  },
  {
    "length" : 1,
    "offset" : 52,
    "type" : "source.lang.swift.syntaxtype.string_interpolation_anchor"
  },
  {
    "length" : 5,
    "offset" : 53,
    "type" : "source.lang.swift.syntaxtype.identifier"
  },
  {
    "length" : 1,
    "offset" : 58,
    "type" : "source.lang.swift.syntaxtype.string_interpolation_anchor"
  },
  {
    "length" : 1,
    "offset" : 59,
    "type" : "source.lang.swift.syntaxtype.string"
  }
]
```

Searching by (53, 5) would return only the first token, because the second one isn't in the range. However, the 3rd one is and should be included in the result.